server.go

```diff
@@ -1246,7 +1246,7 @@ func (s *server) handleDonePeerMsg(state *peerState, sp *serverPeer) {
 		if !sp.Inbound() && sp.VersionKnown() {
 			state.outboundGroups[addrmgr.GroupKey(sp.NA())]--
 		}
-		if sp.persistent && sp.connReq != nil {
+		if (!sp.Inbound() || sp.persistent) && sp.connReq != nil {
```

Comment:
Good point.  I'll update it shortly.  I assume you meant if `sp.persistent` is true it implies `!sp.Inbound()` is true.